src/command_helpers.rs

```diff
@@ -41,83 +40,51 @@ impl CargoOutput {
         }
     }
 
-    pub(crate) fn print_thread(&self) -> Result<Option<PrintThread>, Error> {
-        self.warnings.then(PrintThread::new).transpose()
+    fn stdio_for_warnings(&self) -> Stdio {
+        if self.warnings {
+            Stdio::piped()
+        } else {
+            Stdio::null()
+        }
     }
 }
 
-pub(crate) struct PrintThread {
-    handle: Option<JoinHandle<()>>,
-    pipe_writer: Option<File>,
+#[cfg(feature = "parallel")]
+pub(crate) fn reader_for_stderr_as_warnings(
+    child: &mut Child,
+) -> Option<BufReader<std::process::ChildStderr>> {
+    child
+        .stderr
+        .take()
+        .map(|stderr| BufReader::with_capacity(4096, stderr))
```

Comment:
I feel like we don't need 4096 B for the buffer, IMO I rarely see any warning for my project using cc-rs indirectly, and even if it does it's unlikely to print 4096B of characters.

If that happens, it will likely be a compilation error in C++ which dumps a lot of messages due to overloading.

It's so rare and usually signals a failure, so I don't think we need to optimize for that particular use case.

We should instead tune it down to save memory, allocating a whole page is going to be quite expensive since it's large.

I'd recommend something like 100B, should be good enough.

Reply:
I'll switch this to `BufReader::new` instead of explicitly setting a capacity.